### DISCUSSION TITLE
Playwright page.evaluate error causing scenario to hang

### DISCUSSION THREAD

**Discussion by tchambers97** [artilleryio/artillery#1648] [Artillery]
Hello,
I'm encountering this error rather frequently during a playwright test:
`worker error, id: 1 page.evaluate: Target closed
    at Page.<anonymous> (node_modules/artillery-engine-playwright/index.js:71:52) {
  name: 'Error'
}`

The error causes the scenario to hang and never quit. A `try...catch` around that block would suppress the error. The playwright command I'm running is `await page.locator('text=/add to cart|add to bag/i').waitFor();`

Any idea what the problem could be? Or I could just put up a PR with the `try...catch` ?

**Comment by hassy** [comment]
hi @tchambers97 👋 a `try...catch` sounds reasonable! is there any pattern to when that error tends to be raised?

the engine could possible report those errors as a custom counter metric to the user via the [custom metrics API](https://www.artillery.io/docs/guides/guides/extension-apis#tracking-custom-metrics)

**Reply by tchambers97** [reply]
No pattern that I can see. I ran it with `headless: false` and it just closes the page. Happens about 2 out of 10 runs. 

I also changed the `waitFor()` to a `click()` and that seemed to resolve it.

**Reply by hassy** [reply]
Okay, interesting. It seems the default timeout for `waitFor()` is 30 seconds ([docs](https://playwright.dev/docs/api/class-locator#locator-wait-for)) I wonder if it has to do with resource usage, i.e. if too many VUs (Playwright instances) are being run per Fargate container (or locally) causing the browsers to struggle to render the page.

**Reply by tchambers97** [reply]
I don't think so. I've seen it happen with only one VU.
